DeezyMatch/utils.py

```diff
@@ -646,34 +651,38 @@ def log_plotter(path2log, dataset="DEFAULT"):
         plt.text(valid_arr[min_valid_arg, 0]*1.05, min(min(valid_arr[:, 3]), min(valid_arr[:, 4]), min(train_arr[:, 3]), min(train_arr[:, 4])), 
-                 f"Epoch: {min_valid_arg}, Prec/Recall: {valid_arr[min_valid_arg, 3]}/{valid_arr[min_valid_arg, 4]}", fontsize=12, color="r")
+                 f"Epoch: {min_valid_arg+1}, Prec/Recall: {valid_arr[min_valid_arg, 3]}/{valid_arr[min_valid_arg, 4]}", fontsize=12, color="r")
     plt.xlabel("Epoch", size=18)
     plt.ylabel("Precision/Recall", size=18)
     plt.legend(fontsize=14, loc=4)
-    plt.xticks(size=14)
+    plt.xticks(train_arr[:, 0], train_arr[:, 0].astype(np.integer), size=14)
     plt.yticks(size=14)
     plt.grid()
     
-    plt.subplot(3, 2, 5)
-    plt.title(f"Dataset: {dataset}\nTotal time: {total_time}s, Ave. Time / epoch: {total_time/(len(time_arr)-1):.3f}s", size=16)
-    plt.plot(train_arr[1:, 0], diff_time, c="k", lw=2)
+    plt.figtext(0.5, 0.25, f"Dataset: {dataset}\nTotal time: {total_time}s\nAve. Time / epoch: {total_time/(len(time_arr)-1):.3f}s", 
+                ha="center", fontsize=14, bbox={"facecolor": "beige", "alpha":0.5, "pad":5})
+
+    # >>>>>> Plot time per epoch, commented out for now, we provide a text summary
+    # plt.subplot(3, 2, 5)
+    # plt.title(f"Dataset: {dataset}\nTotal time: {total_time}s, Ave. Time / epoch: {total_time/(len(time_arr)-1):.3f}s", size=16)
+    # plt.plot(train_arr[1:, 0], diff_time, c="k", lw=2)
 
     # If min_valid_arg is 0 (the first model has the lowest valid loss)
     # Increment min_valid_arg for Time as we use cumsum (lose one point in the plot)
-    if min_valid_arg == 0:
-        min_valid_arg += 1
+    # if min_valid_arg == 0:
+    #     min_valid_arg += 1
+
+    # if plot_valid:
+    #     plt.axvline(valid_arr[min_valid_arg, 0], 0, 1, ls="--", c="k")
+    #     plt.text(valid_arr[min_valid_arg, 0]*1.05, min(diff_time)*0.98, 
+    #              f"Epoch: {min_valid_arg+1}, Time to solution: {np.cumsum(diff_time[:min_valid_arg])[-1]}s", fontsize=12, color="r")
+    # plt.xlabel("Epoch", size=18)
+    # plt.ylabel("Time (each epoch) / sec", size=18)
+    # plt.xticks(train_arr[:, 0], train_arr[:, 0].astype(np.integer), size=14)
+    # plt.yticks(size=14)
+    # plt.ylim(min(diff_time)*0.97)
+    # plt.grid()
 
-    if plot_valid:
-        plt.axvline(valid_arr[min_valid_arg, 0], 0, 1, ls="--", c="k")
-        plt.text(valid_arr[min_valid_arg, 0]*1.05, min(diff_time)*0.98, 
-                 f"Epoch: {min_valid_arg}, Time to solution: {np.cumsum(diff_time[:min_valid_arg])[-1]}s", fontsize=12, color="r")
-    plt.xlabel("Epoch", size=18)
-    plt.ylabel("Time (each epoch) / sec", size=18)
-    plt.xticks(size=14)
-    plt.yticks(size=14)
-    plt.ylim(min(diff_time)*0.97)
-    plt.grid()
-    
     plt.tight_layout()
     path2fig = os.path.join(path2fig_dir, f"log_{dataset}.png")
-    plt.savefig(path2fig, dpi=300)
+    plt.savefig(path2fig, dpi=300, bbox_inches='tight', pad_inches=0)
```

Comment:
Hi @kasra-hosseini, just a small thing: the legend sometimes is on top of the best epoch's performance. For example, for `/deezy_datadrive/mariona/DeezyMatch/models/finetuned_test50_trivial_to_ocrgb1900_fc/log.txt`:
![log_test001](https://user-images.githubusercontent.com/46483603/89059148-90e91a80-d358-11ea-9c09-1defefd80ffa.png)

Reply:
I changed the figure, see: https://github.com/Living-with-machines/DeezyMatch/tree/feature/50-alias-detection-on-fly#train-a-new-model

Reply:
It looks great, thanks!